fmradioiriscontrol.cpp

```diff
@@ -78,10 +76,11 @@ FMRadioIrisControl::FMRadioIrisControl()
 FMRadioIrisControl::~FMRadioIrisControl()
 {
     m_timer->stop();
-    SetCtrl(V4L2_CID_PRIVATE_IRIS_STATE, 0);
-    if (m_eventListenerThread > 0) {
-        pthread_join(m_eventListenerThread, NULL);
-    }
+    m_workerThread->setQuit();
+    SetCtrl(V4L2_CID_PRIVATE_IRIS_STATE, 0); // assuming this wakes blocking ioctl()
+    m_workerThread->wait();
+    delete m_workerThread;
+    close(m_fd);
```

Comment:
This fails to build "fmradioiriscontrol.cpp:83:15: error: 'close' was not declared in this scope".